### PR TITLE
Fix Content pack install modal have two superposed modal element

### LINKED ISSUE #15752: Content Packs page "Show Rows" button not properly right-aligned
Untitled

## Expected Behavior

Show Rows button should be all the way to the right edge of the screen like the Show Rows buttons on the Pipelines page. 

## Current Behavior

The top and bottom "Show Rows" buttons are not fully aligned to the right side of the page. As far as I can tell this only occurs on the Content Packs page, and does not occur on other pages with this button like the Pipelines page.

I've confirmed this occurs on at least the following resolutions:
- 1920x1080
- 2560x1080
- 3440x1440

There appears to be quite a few differences in the CSS for the relevant HTML <div> tags between the Content Packs page and the Pipelines pages:

Content Packs page: 
image
Pipelines page: 
image

## Steps to Reproduce (for bugs)

1. Load the Content Packs page

## Your Environment

* Graylog Version: 5.1.1
* Java Version: bundled
* Elasticsearch Version: n/a
* MongoDB Version: n/a
* Operating System: n/a
* Browser version: Firefox v114.0 / Chrome v114.0.5735.106

### LINKED ISSUE #16834: UI problem when installing content pack from content pack list page
## Description

When installing a content pack from the content pack list page, there are two modals opened instead of just one:

image

Installing from a content pack page doesn't do that.

image

## Steps to Reproduce (for bugs)

1. Navigate to "System / Content Packs"
2. Press the "Install" button for a content pack in the list
3. Observe the two modals showing up
4. Navigate to "System / Content Packs"
5. Open the details page for a content pack
6. Select Actions :arrow_right: Install in the "Versions" section
7. Observe a single modal showing up

## Your Environment

* Graylog Version: 5.2.0-beta.1

### PR DESCRIPTION
**Pull request by ousmaneo**
This PR fixes the issue with the Content pack installation described in the issue [here](https://github.com/Graylog2/graylog2-server/issues/16834). We also fixed the `Show rows` button alignment.

fix #16834 
fix #15752
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: graylog2-web-interface/src/components/content-packs/ContentPackVersions.tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  onInstall: PropTypes.func,
+};
+
+ContentPackVersions.defaultProps = {
+  onChange: () => {
+  },
+  onDeletePack: () => {
+  },
+  onInstall: () => {
+  },
```
Comment (grotlue) [inline graylog2-web-interface/src/components/content-packs/ContentPackVersions.tsx:77]:
```suggestion
  onChange: () => {},
  onDeletePack: () => {},
  onInstall: () => {},
```

File: graylog2-web-interface/src/components/content-packs/ContentPacksList.tsx [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+import PropTypes from 'prop-types';
+import styled, { css } from 'styled-components';
+
+import {
+  Col,
+  Row,
+} from 'components/bootstrap';
+import {
+  Pagination, PageSizeSelect, NoSearchResult, NoEntitiesExist,
+} from 'components/common';
```
Comment (grotlue) [inline graylog2-web-interface/src/components/content-packs/ContentPacksList.tsx:28]:
```suggestion
import { Col, Row } from 'components/bootstrap';
import { Pagination, PageSizeSelect, NoSearchResult, NoEntitiesExist } from 'components/common';
```

File: graylog2-web-interface/src/components/content-packs/ContentPacksList.tsx [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+                           onDeletePack={onDeletePack}
+                           onInstall={onInstall} />
+    ));
+  };
+
+  const filterContentPacks = (filteredItems: Array<ContentPackInstallation>) => {
+    setFilteredContentPacks(filteredItems);
+  };
+
+  const itemsShownChange = (pageSize: number) => {
```
Comment (grotlue) [inline graylog2-web-interface/src/components/content-packs/ContentPacksList.tsx:73]: Absolute nit, but we could try to stick to one naming pattern for handlers, e.g. call this one `onChangeItemsShown` so it is similar to `onChangePage`

File: graylog2-web-interface/src/components/content-packs/ContentPacksList.tsx [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+  const pageSizeSelect = (
+    <StyledPageSizeSelect onChange={itemsShownChange}
+                          pageSize={paginationOption.perPage}
+                          pageSizes={[10, 25, 50, 100]} />
+  );
+
+  const noContentMessage = contentPacks.length <= 0
+    ? <NoEntitiesExist>No content packs found. Please create or upload one</NoEntitiesExist>
+    : <NoSearchResult>No matching content packs have been found</NoSearchResult>;
+  const content = filteredContentPacks.length <= 0
```
Comment (grotlue) [inline graylog2-web-interface/src/components/content-packs/ContentPacksList.tsx:98]:
```suggestion
    : <NoSearchResult>No matching content packs have been found</NoSearchResult>;
    
  const content = filteredContentPacks.length <= 0
```

File: graylog2-web-interface/src/components/content-packs/Types.ts [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+  name: string,
+  parameters?: Array<any>,
+  rev: number,
+  server_version: string,
+  summary: string,
+  url: string,
+  v: number,
+  vendor: string,
+}
+export type ContentPackVersionsType = {
```
Comment (grotlue) [inline graylog2-web-interface/src/components/content-packs/Types.ts:32]:
```suggestion
}

export type ContentPackVersionsType = {
```

File: graylog2-web-interface/src/components/content-packs/Types.ts [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+  contentPacks: Array<ContentPackInstallation>,
+  latest: ContentPackInstallation,
+  latestRevision: number,
+  revisions: Array<number>
+}
+export type ContentPackEntity = {
```
Comment (grotlue) [inline graylog2-web-interface/src/components/content-packs/Types.ts:38]:
```suggestion
}

export type ContentPackEntity = {
```

File: graylog2-web-interface/src/components/content-packs/components/ContentPackListItem.test.tsx [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+  };
+
+  it('render content pack item', async () => {
+    render(
+      <ContentPackListItem pack={pack}
+                           contentPackMetadata={metadata}
+                           onDeletePack={() => {
+                           }}
+                           onInstall={() => {
+                           }} />);
```
Comment (grotlue) [inline graylog2-web-interface/src/components/content-packs/components/ContentPackListItem.test.tsx:49]:
```suggestion
                           onDeletePack={() => {}}
                           onInstall={() => {}} />);
```

File: graylog2-web-interface/src/components/content-packs/components/ContentPackListItem.test.tsx [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+
+  it('delete content pack item version', async () => {
+    const deleteFn = jest.fn();
+
+    render(
+      <ContentPackListItem pack={pack}
+                           contentPackMetadata={metadata}
+                           onDeletePack={deleteFn}
+                           onInstall={() => {
+                           }} />);
```
Comment (grotlue) [inline graylog2-web-interface/src/components/content-packs/components/ContentPackListItem.test.tsx:62]:
```suggestion
                           onInstall={() => {}} />);
```

File: graylog2-web-interface/src/components/content-packs/components/ContentPackListItem.test.tsx [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+
+  it('install content pack item version', async () => {
+    const deleteFn = jest.fn();
+
+    render(
+      <ContentPackListItem pack={pack}
+                           contentPackMetadata={metadata}
+                           onDeletePack={deleteFn}
+                           onInstall={() => {
+                           }} />);
```
Comment (grotlue) [inline graylog2-web-interface/src/components/content-packs/components/ContentPackListItem.test.tsx:77]:
```suggestion
                           onInstall={() => {}} />);
```

File: graylog2-web-interface/src/components/content-packs/components/ContentPackVersionItem.tsx [hunk 10]
```diff
@@ -0,0 +1,153 @@
...
+  onInstall: PropTypes.func,
+};
+
+ContentPackVersionItem.defaultProps = {
+  onChange: () => {
+  },
+  onDeletePack: () => {
+  },
+  onInstall: () => {
+  },
```
Comment (grotlue) [inline graylog2-web-interface/src/components/content-packs/components/ContentPackVersionItem.tsx:150]:
```suggestion
  onChange: () => {},
  onDeletePack: () => {},
  onInstall: () => {},
```

**Review by grotlue — APPROVED** [review]
Looks good! All code comments are mainly nit picking. Do I have a different linter config? The empty functions on multiple lines appear unusual to me.

✅ Content pack install shows one modal when opened from Content Pack Overview List
✅ Content pack install shows one modal when opened from Content Pack Detail View
✅ Install modal can be closed by clicking the `X` in Content Pack Overview List
✅ Install modal can be closed by clicking the `X` in Content Pack Detail View
✅ Install modal can be closed by clicking `Cancel` in Content Pack Overview List
✅ Install modal can be closed by clicking `Cancel` in Content Pack Detail View
✅ Content pack is installed after clicking `Install` in Content Pack Overview List
✅ Content pack is installed after clicking `Install` in Content Pack Detail View
✅ Show Rows buttons are properly aligned above and below the Content Pack Overview List

Screenshot 2023-11-10 at 12 57 45
Screenshot 2023-11-10 at 13 05 01

**Comment by ousmaneo** [comment]
> Looks good! All code comments are mainly nit picking. Do I have a different linter config? The empty functions on multiple lines appear unusual to me.


Thank you for reviewing. Intellij is doing funny stuff with the linter. I fixed it.

**Review by grotlue — APPROVED** [review]
Thank you, looking good!

**Pull request merged**